13-multiplecomponents-lab/src/03-shoppinglist-filters.jsx

```diff
@@ -5,3 +5,131 @@
  *  - Ability to show only products matching a given name
  *  - Ability to show only products that were fully bought
  */
+
+/**
+ * Created by ofek_am on 18/12/2016.
+ */
+import React from 'react';
+import _ from 'underscore';
+
+export default class ShoplistFilter extends React.Component{
+
+    static propTypes = {
+
+    };
+
+    static defaultProps = {
+
+    };
+
+    id=0;
+
+    items=[];
+
+    constructor(props){
+        super(props);
+        this.add = this.add.bind(this);
+        this.remove = this.remove.bind(this);
+        this.buy = this.buy.bind(this);
+        this.filter = this.filter.bind(this);
+    }
+
+    state={
+        items:[],
+    }
+
+    add(e){
+        let name=document.getElementById('name').value;
```

Comment:
never access dom directly unless you really need this for 3rd party integration or similar. There is no reason you shouldn't store current item properties in state and reuse them when adding item.
Another option is to use [refs](https://facebook.github.io/react/docs/refs-and-the-dom.html) to access dom.